assets/js/script.js

```diff
@@ -181,61 +337,107 @@ function attractions(cityLon, cityLat, restaurantOneLon, restaurantOneLat, resta
+            waypoints: [{ location: restData.restTwo.lat + ", " + restData.restTwo.lon }, { location: attractData.eventTwo.lat + ", " + attractData.eventTwo.lon }],
+
+            travelMode: google.maps.TravelMode.DRIVING
+        },
+        (response, status) => {
+            if (status === "OK") {
+                directionsRenderer.setDirections(response);
+            } else {
+                window.alert("Directions request failed due to " + status);
+            }
+        }
+    );
+}
+
+// function createMap() {
+// clear out old script if there is some
+// if (document.getElementById("google-maps-api")) {
+//     document.getElementById("google-maps-api").remove();
+//     $('head').children('script').remove();
+//     $('head').children('style').addClass("google-style-api");
+// $('.google-style-api').remove();
+// }
+
+// Create the script tag, set the appropriate attributes
+// var script = document.createElement('script');
+// script.src = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyCv_iF_YniNOH9mI6WvJc66w5bo3_PXXCg&callback=initMap';
+// script.defer = false;
+// script.id = "google-maps-api";
+
+// Append the 'script' element to 'head'
+// mapScriptContainer.appendChild(script);
+
+// initates function to create google map
+
+
+// }
+// ------------------------------------------------------------------------------------------------------------------------------------- //
+
+
+// ------------------------------------------------------------------------------------------------------------------------------------- //
+// ------ generate Itinerary  ------ //
+// this function pulls recently fetched data and contains it into an object
+// the object includes the city name, waypoints, fetched restaurants/event locations, and respective urls
+// calls display Itinerary function
+// ------------------------------------------------------------------------------------------------------------------------------------- //
+function generateItinerary() {
+    var displayCity = cityData.userInput.searchTerm.toUpperCase();
+    var cityLatCoord = cityData.cityCoord.lat;
+    var cityLonCoord = cityData.cityCoord.lon;
+    var wayPointArray = ["A", "B", "C", "D"];
+    var placeArray = [restData.restOne.restName, attractData.eventOne.eventName, restData.restTwo.restName, attractData.eventTwo.eventName];
+    var urlArray = [restData.restOne.url, attractData.eventOne.url, restData.restTwo.url, attractData.eventTwo.url];
+    var latArray = [restData.restOne.lat, attractData.eventOne.lat, restData.restTwo.lat, attractData.eventTwo.lat];
+    var lonArray = [restData.restOne.lon, attractData.eventOne.lon, restData.restTwo.lon, attractData.eventTwo.lon];
+
+    var itineraryObject = {
+        "city": displayCity, "cityLat": cityLatCoord, "cityLong": cityLonCoord, "waypoint": wayPointArray, "place": placeArray, "url": urlArray, "lat": latArray,
+        "long": lonArray, "restData": restData, "cityData": cityData, "attractData": attractData
```

Comment:
We don't need to save restData, cityData, and attractData to local storage anymore